src/core/components/courses/CompanyTable.vue

```diff
@@ -45,16 +67,60 @@ export default {
     ]);
     const pagination = ref({ rowsPerPage: 0, sortBy: 'name' });
 
-    const companiesCount = computed(() => companies.value.length);
+    const course = computed(() => $store.state.course.course);
+
+    const {
+      selectedCompany,
+      companyAdditionModal,
+      companyModalLoading,
+      companyValidation,
+      resetCompanyAdditionModal,
+      companiesCount,
+      getPotentialCompanies,
+      companyOptions,
+      openCompanyAdditionModal,
+    } = useCompanies(companies, course);
+
+    const addCompany = async () => {
```

Comment:
Pourquoi avoir laissé cette méthode hors de la composables ?

Reply:
à cause du emit, j'ai pas l'impression qu'on puisse l'utiliser dans la composable, tu as une solution pour ça ?

Reply:
J'ai l'impression qu'il y a des moyens de faire, a tenter : https://stackoverflow.com/questions/72394953/how-to-fire-an-emit-event-from-vue-composable